autotag.go

```diff
@@ -159,6 +159,30 @@ func NewRepo(cfg GitRepoConfig) (*GitRepo, error) {
 		return nil, err
 	}
 
+	if cfg.Branch == "" {
+		branches, err := repo.Branches()
+		if err != nil {
+			return nil, err
+		}
+		for _, b := range branches {
+			if b == "master" {
+				cfg.Branch = "master"
+				break
+			}
+		}
+		if cfg.Branch == "" {
+			for _, b := range branches {
+				if b == "main" {
+					cfg.Branch = "main"
+					break
+				}
+			}
+			if cfg.Branch == "" {
+				return nil, fmt.Errorf("no master or main branch found")
+			}
+		}
```

Comment:
```suggestio
		branches, err := repo.Branches()
		if err != nil {
			return nil, err
		}
		hasMain := false
		for _, b := range branches {
			if b == "master" {
                                cfg.Branch = "master"
			}
			if b == "main" {
				hasMain := true        
		        }
		}

		if hasMain == true {
			cfg.Branch = "main"
                }
	
                if cfg.Branch == "" {
			return nil, fmt.Errorf("no master or main branch found")
		}
```

Reply:
this should select master if it exists, and prefer main if it exists. removes nesting and IMO is slightly more declarative.  Could reverse master/main if we want the reverse , but practically speaking folks will be moving from master to main and generally not the other way around.

Reply:
I can do this without the temp variable at all, if overwriting cfg.Branch is tolerable (see latest/third commit, added today).